Eclipse generate throws InvocationTargetException if context.xml is present in template sets and root

### Expected behavior

As a CobiGen Eclipse user, I want CobiGen to cancel the generate process and display a proper error message when a conflict between the new and old template directory structure exists.

### Actual behavior
Eclipse tries to load the templates and throws an InvocationTargetException.

### Steps to reproduce (bug) / Use Case of feature request (enhancement)

1. Create a simple maven project with an example OpenApi file to generate from
2. Import the templates from the latest master into your workspace
3. Copy a random context.xml from one of the template sets into the root of CobiGen_Templates
4. Start a generate from the OpenApi file

### Comments/Hints:
The HealthCheck of Eclipse will detect this issue and display the expected error message.


### Affected version:

* Eclipse Plug-in 2021.12.007-SNAPSHOT

This seems to be a general problem of the error dialog of the Eclipse generate command and does not only refer to the new configuration structure.
With Cobigen 2021.12.006 and an invalid 'old' configuration structure (e.g. without context.xml) the dialog does not show a correct message either

![eclipse_error](https://user-images.githubusercontent.com/69513769/157076724-3bee1bb8-f4ec-4220-ae2e-3ed186cd7803.PNG)

```
InvocationTargetException
java.lang.reflect.InvocationTargetException
  java.lang.reflect.InvocationTargetException
  	at org.eclipse.jface.operation.ModalContext.run(ModalContext.java:395)
  	at org.eclipse.jface.dialogs.ProgressMonitorDialog.run(ProgressMonitorDialog.java:468)
  	at com.devonfw.cobigen.eclipse.workbenchcontrol.handler.GenerateHandler.execute(GenerateHandler.java:42)
  	at org.eclipse.ui.internal.handlers.HandlerProxy.execute(HandlerProxy.java:283)
  	at org.eclipse.ui.internal.handlers.E4HandlerProxy.execute(E4HandlerProxy.java:97)
  	at jdk.internal.reflect.GeneratedMethodAccessor114.invoke(Unknown Source)
  	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
  	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
  	at org.eclipse.e4.core.internal.di.MethodRequestor.execute(MethodRequestor.java:58)
  	at org.eclipse.e4.core.internal.di.InjectorImpl.invokeUsingClass(InjectorImpl.java:317)
  	at org.eclipse.e4.core.internal.di.InjectorImpl.invoke(InjectorImpl.java:251)
  	at org.eclipse.e4.core.contexts.ContextInjectionFactory.invoke(ContextInjectionFactory.java:173)
  	at org.eclipse.e4.core.commands.internal.HandlerServiceHandler.execute(HandlerServiceHandler.java:156)
  	at org.eclipse.core.commands.Command.executeWithChecks(Command.java:488)
  	at org.eclipse.core.commands.ParameterizedCommand.executeWithChecks(ParameterizedCommand.java:487)
  	at org.eclipse.e4.core.commands.internal.HandlerServiceImpl.executeHandler(HandlerServiceImpl.java:213)
  	at org.eclipse.e4.ui.workbench.renderers.swt.HandledContributionItem.executeItem(HandledContributionItem.java:438)
  	at org.eclipse.e4.ui.workbench.renderers.swt.AbstractContributionItem.handleWidgetSelection(AbstractContributionItem.java:449)
  	at org.eclipse.e4.ui.workbench.renderers.swt.AbstractContributionItem.lambda$2(AbstractContributionItem.java:475)
  	at org.eclipse.swt.widgets.EventTable.sendEvent(EventTable.java:89)
  	at org.eclipse.swt.widgets.Display.sendEvent(Display.java:4243)
  	at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1060)
  	at org.eclipse.swt.widgets.Display.runDeferredEvents(Display.java:4060)
  	at org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:3632)
  	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine$5.run(PartRenderingEngine.java:1154)
  	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:338)
  	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.run(PartRenderingEngine.java:1045)
  	at org.eclipse.e4.ui.internal.workbench.E4Workbench.createAndRunUI(E4Workbench.java:155)
  	at org.eclipse.ui.internal.Workbench.lambda$3(Workbench.java:644)
  	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:338)
  	at org.eclipse.ui.internal.Workbench.createAndRunWorkbench(Workbench.java:551)
  	at org.eclipse.ui.PlatformUI.createAndRunWorkbench(PlatformUI.java:156)
  	at org.eclipse.ui.internal.ide.application.IDEApplication.start(IDEApplication.java:152)
  	at org.eclipse.equinox.internal.app.EclipseAppHandle.run(EclipseAppHandle.java:203)
  	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.runApplication(EclipseAppLauncher.java:136)
  	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.start(EclipseAppLauncher.java:104)
  	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:401)
  	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:255)
  	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
  	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
  	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
  	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
  	at org.eclipse.equinox.launcher.Main.invokeFramework(Main.java:659)
  	at org.eclipse.equinox.launcher.Main.basicRun(Main.java:596)
  	at org.eclipse.equinox.launcher.Main.run(Main.java:1467)
  	at org.eclipse.equinox.launcher.Main.main(Main.java:1440)
  Caused by: org.eclipse.swt.SWTException: Invalid thread access
  	at org.eclipse.swt.SWT.error(SWT.java:4907)
  	at org.eclipse.swt.SWT.error(SWT.java:4822)
  	at org.eclipse.swt.SWT.error(SWT.java:4793)
  	at org.eclipse.swt.widgets.Display.error(Display.java:1272)
  	at org.eclipse.swt.widgets.Display.checkDevice(Display.java:827)
  	at org.eclipse.swt.widgets.Display.getActiveShell(Display.java:1490)
  	at com.devonfw.cobigen.eclipse.common.tools.ExceptionHandler.openInvalidConfigurationErrorDialog(ExceptionHandler.java:77)
  	at com.devonfw.cobigen.eclipse.common.tools.ExceptionHandler.handle(ExceptionHandler.java:38)
  	at com.devonfw.cobigen.eclipse.workbenchcontrol.handler.GenerateJob.run(GenerateJob.java:106)
  	at org.eclipse.jface.operation.ModalContext$ModalContextThread.run(ModalContext.java:122)

```

--- a/cobigen-eclipse/cobigen-eclipse-test/src/main/java/com/devonfw/cobigen/eclipse/test/ExceptionHandlingTest.java
+++ b/cobigen-eclipse/cobigen-eclipse-test/src/main/java/com/devonfw/cobigen/eclipse/test/ExceptionHandlingTest.java
@@ -18,7 +18,6 @@ import org.junit.runner.RunWith;
 import com.devonfw.cobigen.eclipse.common.constants.external.ResourceConstants;
 import com.devonfw.cobigen.eclipse.test.common.SystemTest;
 import com.devonfw.cobigen.eclipse.test.common.junit.TmpMavenProjectRule;
-import com.devonfw.cobigen.eclipse.test.common.swtbot.AllJobsAreFinished;
 import com.devonfw.cobigen.eclipse.test.common.utils.EclipseCobiGenUtils;
 import com.devonfw.cobigen.eclipse.test.common.utils.EclipseUtils;
 
@@ -73,7 +72,6 @@ public class ExceptionHandlingTest extends SystemTest {
 
     // execute CobiGen
     EclipseCobiGenUtils.processCobiGenWithExpectedError(bot, javaClassItem, "Invalid context configuration!");
-    bot.waitUntil(new AllJobsAreFinished(), 10000);
   }
 
 }

--- a/cobigen-eclipse/cobigen-eclipse-test/src/main/java/com/devonfw/cobigen/eclipse/test/ExceptionHandlingTest.java
+++ b/cobigen-eclipse/cobigen-eclipse-test/src/main/java/com/devonfw/cobigen/eclipse/test/ExceptionHandlingTest.java
@@ -11,13 +11,11 @@ import org.eclipse.swtbot.eclipse.finder.widgets.SWTBotView;
 import org.eclipse.swtbot.swt.finder.junit.SWTBotJunit4ClassRunner;
 import org.eclipse.swtbot.swt.finder.widgets.SWTBotTreeItem;
 import org.junit.BeforeClass;
-import org.junit.Rule;
 import org.junit.Test;
 import org.junit.runner.RunWith;
 
 import com.devonfw.cobigen.eclipse.common.constants.external.ResourceConstants;
 import com.devonfw.cobigen.eclipse.test.common.SystemTest;
-import com.devonfw.cobigen.eclipse.test.common.junit.TmpMavenProjectRule;
 import com.devonfw.cobigen.eclipse.test.common.utils.EclipseCobiGenUtils;
 import com.devonfw.cobigen.eclipse.test.common.utils.EclipseUtils;
 
@@ -29,10 +27,6 @@ public class ExceptionHandlingTest extends SystemTest {
 
   /** Root path of the Test Resources */
   private static final String resourcesRootPath = "src/main/resources/ExceptionHandlingTest/";
-
-  /** Rule for creating temporary {@link IJavaProject}s per test. */
-  @Rule
-  public TmpMavenProjectRule tmpMavenProjectRule2 = new TmpMavenProjectRule();
 
   /**
    * Setup workbench appropriately for tests

--- a/cobigen-eclipse/cobigen-eclipse-test/src/main/java/com/devonfw/cobigen/eclipse/test/ExceptionHandlingTest.java
+++ b/cobigen-eclipse/cobigen-eclipse-test/src/main/java/com/devonfw/cobigen/eclipse/test/ExceptionHandlingTest.java
@@ -8,20 +8,24 @@ import org.eclipse.core.runtime.NullProgressMonitor;
 import org.eclipse.jdt.core.IJavaProject;
 import org.eclipse.jdt.ui.JavaUI;
 import org.eclipse.swtbot.eclipse.finder.widgets.SWTBotView;
+import org.eclipse.swtbot.swt.finder.junit.SWTBotJunit4ClassRunner;
 import org.eclipse.swtbot.swt.finder.widgets.SWTBotTreeItem;
 import org.junit.BeforeClass;
 import org.junit.Rule;
 import org.junit.Test;
+import org.junit.runner.RunWith;
 
 import com.devonfw.cobigen.eclipse.common.constants.external.ResourceConstants;
 import com.devonfw.cobigen.eclipse.test.common.SystemTest;
 import com.devonfw.cobigen.eclipse.test.common.junit.TmpMavenProjectRule;
+import com.devonfw.cobigen.eclipse.test.common.swtbot.AllJobsAreFinished;
 import com.devonfw.cobigen.eclipse.test.common.utils.EclipseCobiGenUtils;
 import com.devonfw.cobigen.eclipse.test.common.utils.EclipseUtils;
 
 /**
  * Test suite for exception handling issues.
  */
+@RunWith(SWTBotJunit4ClassRunner.class)
 public class ExceptionHandlingTest extends SystemTest {
 
   /** Root path of the Test Resources */
@@ -69,6 +73,7 @@ public class ExceptionHandlingTest extends SystemTest {
 
     // execute CobiGen
     EclipseCobiGenUtils.processCobiGenWithExpectedError(bot, javaClassItem, "Invalid context configuration!");
+    bot.waitUntil(new AllJobsAreFinished(), 10000);
   }
 
 }

--- a/cobigen-eclipse/cobigen-eclipse-test/src/main/java/com/devonfw/cobigen/eclipse/test/ExceptionHandlingTest.java
+++ b/cobigen-eclipse/cobigen-eclipse-test/src/main/java/com/devonfw/cobigen/eclipse/test/ExceptionHandlingTest.java
@@ -1,0 +1,74 @@
+package com.devonfw.cobigen.eclipse.test;
+
+import java.io.File;
+
+import org.apache.commons.io.FileUtils;
+import org.eclipse.core.resources.IResource;
+import org.eclipse.core.runtime.NullProgressMonitor;
+import org.eclipse.jdt.core.IJavaProject;
+import org.eclipse.jdt.ui.JavaUI;
+import org.eclipse.swtbot.eclipse.finder.widgets.SWTBotView;
+import org.eclipse.swtbot.swt.finder.widgets.SWTBotTreeItem;
+import org.junit.BeforeClass;
+import org.junit.Rule;
+import org.junit.Test;
+
+import com.devonfw.cobigen.eclipse.common.constants.external.ResourceConstants;
+import com.devonfw.cobigen.eclipse.test.common.SystemTest;
+import com.devonfw.cobigen.eclipse.test.common.junit.TmpMavenProjectRule;
+import com.devonfw.cobigen.eclipse.test.common.utils.EclipseCobiGenUtils;
+import com.devonfw.cobigen.eclipse.test.common.utils.EclipseUtils;
+
+/**
+ * Test suite for exception handling issues.
+ */
+public class ExceptionHandlingTest extends SystemTest {
+
+  /** Root path of the Test Resources */
+  private static final String resourcesRootPath = "src/main/resources/ExceptionHandlingTest/";
+
+  /** Rule for creating temporary {@link IJavaProject}s per test. */
+  @Rule
+  public TmpMavenProjectRule tmpMavenProjectRule2 = new TmpMavenProjectRule();
+
+  /**
+   * Setup workbench appropriately for tests
+   *
+   * @throws Exception test fails
+   */
+  @BeforeClass
+  public static void setupClass() throws Exception {
+
+    EclipseUtils.cleanWorkspace(bot, true);
+    // import the configuration project for this test
+    EclipseUtils.importExistingGeneralProject(bot, new File(resourcesRootPath + "templates").getAbsolutePath());
+    EclipseUtils.updateMavenProject(bot, ResourceConstants.CONFIG_PROJECT_NAME);
+  }
+
+  /**
+   * Tests for a conflict of old and new templates configuration and proper error message dialog
+   *
+   * @throws Exception Test fails
+   */
+  @Test
+  public void testConflictWithTemplateTypesDisplaysErrorDialog() throws Exception {
+
+    // create a new temporary java project and copy java class used as an input for CobiGen
+    String testProjectName = "TestInputProj";
+    IJavaProject project = this.tmpMavenProjectRule.createProject(testProjectName);
+    FileUtils.copyFile(new File(resourcesRootPath + "input/PlainInput.java"),
+        project.getUnderlyingResource().getLocation().append("src/main/java/main/PlainInput.java").toFile());
+    project.getProject().refreshLocal(IResource.DEPTH_INFINITE, new NullProgressMonitor());
+    this.tmpMavenProjectRule.updateProject();
+
+    // expand the new file in the package explorer
+    SWTBotView view = bot.viewById(JavaUI.ID_PACKAGES);
+    SWTBotTreeItem javaClassItem = view.bot().tree().expandNode(testProjectName, "src/main/java", "main",
+        "PlainInput.java");
+    javaClassItem.select();
+
+    // execute CobiGen
+    EclipseCobiGenUtils.processCobiGenWithExpectedError(bot, javaClassItem, "Invalid context configuration!");
+  }
+
+}

--- a/cobigen-eclipse/cobigen-eclipse-test/src/main/java/com/devonfw/cobigen/eclipse/test/common/utils/EclipseCobiGenUtils.java
+++ b/cobigen-eclipse/cobigen-eclipse-test/src/main/java/com/devonfw/cobigen/eclipse/test/common/utils/EclipseCobiGenUtils.java
@@ -141,6 +141,10 @@ public class EclipseCobiGenUtils {
     input.contextMenu("CobiGen").menu("Generate...").click();
     bot.waitUntil(new AnyShellIsActive(expectedErrorTitle), defaultTimeout);
 
+    takeScreenshot(bot, "InvalidConfigurationDialog");
+
+    bot.waitUntil(new AllJobsAreFinished(), 10000);
+
     SWTBotButton finishButton = bot.button(IDialogConstants.OK_LABEL);
     bot.waitUntil(widgetIsEnabled(bot.button()));
     finishButton.click();

--- a/cobigen-eclipse/cobigen-eclipse-test/src/main/java/com/devonfw/cobigen/eclipse/test/common/utils/EclipseCobiGenUtils.java
+++ b/cobigen-eclipse/cobigen-eclipse-test/src/main/java/com/devonfw/cobigen/eclipse/test/common/utils/EclipseCobiGenUtils.java
@@ -47,6 +47,20 @@ public class EclipseCobiGenUtils {
   }
 
   /**
+   * Tries a Generate process with an expected error title.
+   *
+   * @param bot the {@link SWTWorkbenchBot} of the test
+   * @param input input of CobiGen to be selected
+   * @param expectedErrorTitle String of expected error title
+   * @throws Exception test fails
+   */
+  public static void processCobiGenWithExpectedError(SWTWorkbenchBot bot, SWTBotTreeItem input,
+      String expectedErrorTitle) throws Exception {
+
+    processCobiGenWithExpectedError(bot, input, DEFAULT_TIMEOUT, expectedErrorTitle);
+  }
+
+  /**
    * Expands multi layer nodes of following format: node1>node2>finalNode
    *
    * @param bot SWTWorkbenchBot to use
@@ -105,6 +119,29 @@ public class EclipseCobiGenUtils {
       }
     }
     SWTBotButton finishButton = bot.button(IDialogConstants.FINISH_LABEL);
+    bot.waitUntil(widgetIsEnabled(bot.button()));
+    finishButton.click();
+  }
+
+  /**
+   * Tries a Generate process with an expected error title.
+   *
+   * @param bot the {@link SWTWorkbenchBot} of the test
+   * @param input input of CobiGen to be selected
+   * @param defaultTimeout timeout to be overwritten
+   * @param expectedErrorTitle String of expected error title
+   * @throws Exception test fails
+   */
+  public static void processCobiGenWithExpectedError(SWTWorkbenchBot bot, SWTBotTreeItem input, int defaultTimeout,
+      String expectedErrorTitle) throws Exception {
+
+    // Open generation wizard with new file as Input
+    ResourcesPlugin.getWorkspace().build(IncrementalProjectBuilder.FULL_BUILD, new NullProgressMonitor());
+    bot.waitUntil(new AllJobsAreFinished(), defaultTimeout); // build might take some time
+    input.contextMenu("CobiGen").menu("Generate...").click();
+    bot.waitUntil(new AnyShellIsActive(expectedErrorTitle), defaultTimeout);
+
+    SWTBotButton finishButton = bot.button(IDialogConstants.OK_LABEL);
     bot.waitUntil(widgetIsEnabled(bot.button()));
     finishButton.click();
   }

--- a/cobigen-eclipse/cobigen-eclipse-test/src/main/java/com/devonfw/cobigen/eclipse/test/common/utils/EclipseCobiGenUtils.java
+++ b/cobigen-eclipse/cobigen-eclipse-test/src/main/java/com/devonfw/cobigen/eclipse/test/common/utils/EclipseCobiGenUtils.java
@@ -143,8 +143,6 @@ public class EclipseCobiGenUtils {
 
     takeScreenshot(bot, "InvalidConfigurationDialog");
 
-    bot.waitUntil(new AllJobsAreFinished(), 10000);
-
     SWTBotButton finishButton = bot.button(IDialogConstants.OK_LABEL);
     bot.waitUntil(widgetIsEnabled(bot.button()));
     finishButton.click();

--- a/cobigen-eclipse/cobigen-eclipse-test/src/main/java/com/devonfw/cobigen/eclipse/test/common/utils/EclipseUtils.java
+++ b/cobigen-eclipse/cobigen-eclipse-test/src/main/java/com/devonfw/cobigen/eclipse/test/common/utils/EclipseUtils.java
@@ -187,7 +187,7 @@ public class EclipseUtils {
         }
         break;
       } catch (Exception e) {
-        LOG.debug("An error occured while trying to delete project: {} {}", projectName, e);
+        LOG.debug("An error occured while trying to delete project: {}", projectName, e);
         Thread.sleep(500);
         if (i == maxRetries) {
           LOG.debug("Not able to cleanup the workspace after: {} retries", maxRetries);

--- a/cobigen-eclipse/cobigen-eclipse-test/src/main/java/com/devonfw/cobigen/eclipse/test/common/utils/EclipseUtils.java
+++ b/cobigen-eclipse/cobigen-eclipse-test/src/main/java/com/devonfw/cobigen/eclipse/test/common/utils/EclipseUtils.java
@@ -187,7 +187,7 @@ public class EclipseUtils {
         }
         break;
       } catch (Exception e) {
-        LOG.debug("An error occured while trying to delete project: {}", projectName);
+        LOG.debug("An error occured while trying to delete project: {} {}", projectName, e);
         Thread.sleep(500);
         if (i == maxRetries) {
           LOG.debug("Not able to cleanup the workspace after: {} retries", maxRetries);

--- a/cobigen-eclipse/cobigen-eclipse-test/src/main/resources/ExceptionHandlingTest/input/PlainInput.java
+++ b/cobigen-eclipse/cobigen-eclipse-test/src/main/resources/ExceptionHandlingTest/input/PlainInput.java
@@ -1,0 +1,5 @@
+package main;
+
+public class PlainInput {
+
+}

--- a/cobigen-eclipse/cobigen-eclipse-test/src/main/resources/ExceptionHandlingTest/templates/.project
+++ b/cobigen-eclipse/cobigen-eclipse-test/src/main/resources/ExceptionHandlingTest/templates/.project
@@ -1,0 +1,23 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<projectDescription>
+	<name>CobiGen_Templates</name>
+	<comment></comment>
+	<projects>
+	</projects>
+	<buildSpec>
+		<buildCommand>
+			<name>org.eclipse.jdt.core.javabuilder</name>
+			<arguments>
+			</arguments>
+		</buildCommand>
+		<buildCommand>
+			<name>org.eclipse.m2e.core.maven2Builder</name>
+			<arguments>
+			</arguments>
+		</buildCommand>
+	</buildSpec>
+	<natures>
+		<nature>org.eclipse.m2e.core.maven2Nature</nature>
+		<nature>org.eclipse.jdt.core.javanature</nature>
+	</natures>
+</projectDescription>

--- a/cobigen-eclipse/cobigen-eclipse-test/src/main/resources/ExceptionHandlingTest/templates/logback.xml
+++ b/cobigen-eclipse/cobigen-eclipse-test/src/main/resources/ExceptionHandlingTest/templates/logback.xml
@@ -1,0 +1,17 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!-- This file is for logback classic. The file contains the configuration 
+	for sl4j logging -->
+<configuration>
+
+	<appender name="STDOUT" class="ch.qos.logback.core.ConsoleAppender">
+		<!-- defines the level of logging -->
+		<layout class="ch.qos.logback.classic.PatternLayout">
+			<Pattern>%d{HH:mm:ss.SSS} [%thread] %-5level %logger{36} - %msg%n
+			</Pattern>
+		</layout>
+	</appender>
+
+	<root level="DEBUG">
+		<appender-ref ref="STDOUT" />
+	</root>
+</configuration>

--- a/cobigen-eclipse/cobigen-eclipse-test/src/main/resources/ExceptionHandlingTest/templates/pom.xml
+++ b/cobigen-eclipse/cobigen-eclipse-test/src/main/resources/ExceptionHandlingTest/templates/pom.xml
@@ -1,0 +1,24 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd"
+  xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
+  <modelVersion>4.0.0</modelVersion>
+  <groupId>com.devonfw.cobigen</groupId>
+  <artifactId>templates-oasp</artifactId>
+  <version>1.0.0</version>
+  <packaging>jar</packaging>
+  <name>CobiGen OASP Templates</name>
+
+  <properties>
+    <maven.compiler.source>1.8</maven.compiler.source>
+    <maven.compiler.target>1.8</maven.compiler.target>
+  </properties>
+
+  <dependencies>
+    <dependency>
+      <groupId>javax.ws.rs</groupId>
+      <artifactId>javax.ws.rs-api</artifactId>
+      <version>2.0</version>
+    </dependency>
+  </dependencies>
+
+</project>

--- a/cobigen-eclipse/cobigen-eclipse-test/src/main/resources/ExceptionHandlingTest/templates/src/main/templates/context.xml
+++ b/cobigen-eclipse/cobigen-eclipse-test/src/main/resources/ExceptionHandlingTest/templates/src/main/templates/context.xml
@@ -1,0 +1,7 @@
+<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
+<contextConfiguration xmlns="http://capgemini.com/devonfw/cobigen/ContextConfiguration" version="2.1">
+    <trigger id="test" type="java">
+        <matcher type="fqn" value=".*">
+        </matcher>
+    </trigger>
+</contextConfiguration>

--- a/cobigen-eclipse/cobigen-eclipse-test/src/main/resources/ExceptionHandlingTest/templates/src/main/templates/template1/TestOutput.txt.ftl
+++ b/cobigen-eclipse/cobigen-eclipse-test/src/main/resources/ExceptionHandlingTest/templates/src/main/templates/template1/TestOutput.txt.ftl
@@ -1,0 +1,1 @@
+This is a test

--- a/cobigen-eclipse/cobigen-eclipse-test/src/main/resources/ExceptionHandlingTest/templates/src/main/templates/template1/context.xml
+++ b/cobigen-eclipse/cobigen-eclipse-test/src/main/resources/ExceptionHandlingTest/templates/src/main/templates/template1/context.xml
@@ -1,0 +1,7 @@
+<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
+<contextConfiguration xmlns="http://capgemini.com/devonfw/cobigen/ContextConfiguration" version="2.1">
+    <trigger id="test" type="java">
+        <matcher type="fqn" value=".*">
+        </matcher>
+    </trigger>
+</contextConfiguration>

--- a/cobigen-eclipse/cobigen-eclipse-test/src/main/resources/ExceptionHandlingTest/templates/src/main/templates/template1/templates.xml
+++ b/cobigen-eclipse/cobigen-eclipse-test/src/main/resources/ExceptionHandlingTest/templates/src/main/templates/template1/templates.xml
@@ -1,0 +1,11 @@
+<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
+<templatesConfiguration xmlns="http://capgemini.com/devonfw/cobigen/TemplatesConfiguration" version="2.1">
+    <templates>
+        <template name="t1" destinationPath="TestOutput.txt" templateFile="TestOutput.txt.ftl"/>
+    </templates>
+    <increments>
+        <increment name="i1" description="increment1">
+            <templateRef ref="t1"/>
+        </increment>
+    </increments>
+</templatesConfiguration>

--- a/cobigen-eclipse/cobigen-eclipse/src/com/devonfw/cobigen/eclipse/common/tools/ExceptionHandler.java
+++ b/cobigen-eclipse/cobigen-eclipse/src/com/devonfw/cobigen/eclipse/common/tools/ExceptionHandler.java
@@ -35,7 +35,7 @@ public class ExceptionHandler {
           + "\n If this problem persists please report it to the CobiGen developers.", e);
     } else if (InvalidConfigurationException.class.isAssignableFrom(e.getClass())) {
       LOG.warn("Invalid configuration.", e);
-      openInvalidConfigurationErrorDialog((InvalidConfigurationException) e, getShell(activeShell));
+      openInvalidConfigurationErrorDialog((InvalidConfigurationException) e);
     } else if (GeneratorProjectNotExistentException.class.isAssignableFrom(e.getClass())) {
       LOG.error(
           "The project '{}' containing the configuration and templates is currently not existent. Please create one or check it out from SVN as stated in the user documentation.",
@@ -56,6 +56,7 @@ public class ExceptionHandler {
       LOG.error("An unexpected exception occurred!", e);
       PlatformUIUtil.openErrorDialog("An unexpected exception occurred!", e);
     }
+
   }
 
   /**
@@ -71,21 +72,24 @@ public class ExceptionHandler {
    * Opens up a message dialog for displaying further guidance on context configuration issues.
    *
    * @param e {@link InvalidConfigurationException} occurred
-   * @param activeShell currently active shell
    */
-  private static void openInvalidConfigurationErrorDialog(InvalidConfigurationException e, Shell activeShell) {
+  private static void openInvalidConfigurationErrorDialog(InvalidConfigurationException e) {
 
-    MessageDialog dialog = new MessageDialog(activeShell, "Invalid context configuration!", null,
-        "Any context/templates configuration has been changed into an invalid state "
-            + "OR is simply outdated, if you recently updated CobiGen. "
-            + "For further investigation and automatic upgrade options start CobiGen's Health Check."
-            + "\n\nOriginal error message: " + e.getMessage(),
-        MessageDialog.ERROR, new String[] { "Health Check", "OK" }, 1);
-    dialog.setBlockOnOpen(true);
+    PlatformUIUtil.getWorkbench().getDisplay().syncExec(() -> {
+      MessageDialog dialog = new MessageDialog(Display.getDefault().getActiveShell(), "Invalid context configuration!",
+          null,
+          "Any context/templates configuration has been changed into an invalid state "
+              + "OR is simply outdated, if you recently updated CobiGen. "
+              + "For further investigation and automatic upgrade options start CobiGen's Health Check."
+              + "\n\nOriginal error message: " + e.getMessage(),
+          MessageDialog.ERROR, new String[] { "Health Check", "OK" }, 1);
+      dialog.setBlockOnOpen(true);
 
-    int result = dialog.open();
-    if (result == 0) {
-      new HealthCheckDialog().execute();
-    }
+      int result = dialog.open();
+      if (result == 0) {
+        new HealthCheckDialog().execute();
+      }
+    });
   }
+
 }

--- a/cobigen-eclipse/cobigen-eclipse/src/com/devonfw/cobigen/eclipse/common/tools/ExceptionHandler.java
+++ b/cobigen-eclipse/cobigen-eclipse/src/com/devonfw/cobigen/eclipse/common/tools/ExceptionHandler.java
@@ -35,7 +35,7 @@ public class ExceptionHandler {
           + "\n If this problem persists please report it to the CobiGen developers.", e);
     } else if (InvalidConfigurationException.class.isAssignableFrom(e.getClass())) {
       LOG.warn("Invalid configuration.", e);
-      openInvalidConfigurationErrorDialog((InvalidConfigurationException) e);
+      openInvalidConfigurationErrorDialog((InvalidConfigurationException) e, getShell(activeShell));
     } else if (GeneratorProjectNotExistentException.class.isAssignableFrom(e.getClass())) {
       LOG.error(
           "The project '{}' containing the configuration and templates is currently not existent. Please create one or check it out from SVN as stated in the user documentation.",
@@ -71,11 +71,11 @@ public class ExceptionHandler {
    * Opens up a message dialog for displaying further guidance on context configuration issues.
    *
    * @param e {@link InvalidConfigurationException} occurred
+   * @param activeShell currently active shell
    */
-  private static void openInvalidConfigurationErrorDialog(InvalidConfigurationException e) {
+  private static void openInvalidConfigurationErrorDialog(InvalidConfigurationException e, Shell activeShell) {
 
-    MessageDialog dialog = new MessageDialog(Display.getDefault().getActiveShell(), "Invalid context configuration!",
-        null,
+    MessageDialog dialog = new MessageDialog(activeShell, "Invalid context configuration!", null,
         "Any context/templates configuration has been changed into an invalid state "
             + "OR is simply outdated, if you recently updated CobiGen. "
             + "For further investigation and automatic upgrade options start CobiGen's Health Check."

--- a/cobigen-eclipse/cobigen-eclipse/src/com/devonfw/cobigen/eclipse/common/tools/PlatformUIUtil.java
+++ b/cobigen-eclipse/cobigen-eclipse/src/com/devonfw/cobigen/eclipse/common/tools/PlatformUIUtil.java
@@ -75,17 +75,14 @@ public class PlatformUIUtil {
    */
   public static void openErrorDialog(final String message, final Throwable cause) {
 
-    getWorkbench().getDisplay().syncExec(new Runnable() {
-      @Override
-      public void run() {
+    getWorkbench().getDisplay().syncExec(() -> {
 
-        if (cause == null) {
-          MessageDialog.openError(Display.getDefault().getActiveShell(), CobiGenDialogConstants.DIALOG_TITLE_ERROR,
-              message);
-        } else {
-          LinkErrorDialog.openError(Display.getDefault().getActiveShell(), CobiGenDialogConstants.DIALOG_TITLE_ERROR,
-              message, createMultiStatus(cause));
-        }
+      if (cause == null) {
+        MessageDialog.openError(Display.getDefault().getActiveShell(), CobiGenDialogConstants.DIALOG_TITLE_ERROR,
+            message);
+      } else {
+        LinkErrorDialog.openError(Display.getDefault().getActiveShell(), CobiGenDialogConstants.DIALOG_TITLE_ERROR,
+            message, createMultiStatus(cause));
       }
     });
 

--- a/cobigen-eclipse/cobigen-eclipse/src/com/devonfw/cobigen/eclipse/workbenchcontrol/handler/GenerateJob.java
+++ b/cobigen-eclipse/cobigen-eclipse/src/com/devonfw/cobigen/eclipse/workbenchcontrol/handler/GenerateJob.java
@@ -76,8 +76,8 @@ public class GenerateJob implements IRunnableWithProgress {
       monitor.beginTask("Searching valid triggers...", 1);
       if (!generator.isValidInput(monitor)) {
         LOG.info("No matching Trigger. Exiting generate command.");
-        PlatformUIUtil.getWorkbench().getDisplay()
-            .syncExec(() -> MessageDialog.openInformation(HandlerUtil.getActiveShell(this.event), "No matching Trigger!",
+        PlatformUIUtil.getWorkbench().getDisplay().syncExec(
+            () -> MessageDialog.openInformation(HandlerUtil.getActiveShell(this.event), "No matching Trigger!",
                 "Your current selection is not valid as input for any generation purpose. "
                     + "Please find the specification of valid inputs in the context configuration ('"
                     + ResourceConstants.CONFIG_PROJECT_NAME + "/context.xml')."));
@@ -103,7 +103,10 @@ public class GenerateJob implements IRunnableWithProgress {
         }
       });
     } catch (Throwable e) {
-      ExceptionHandler.handle(e, HandlerUtil.getActiveShell(this.event));
+      PlatformUIUtil.getWorkbench().getDisplay().syncExec(() -> {
+        ExceptionHandler.handle(e, HandlerUtil.getActiveShell(this.event));
+      });
+
     }
 
     MDC.remove(InfrastructureConstants.CORRELATION_ID);

--- a/cobigen-eclipse/cobigen-eclipse/src/com/devonfw/cobigen/eclipse/workbenchcontrol/handler/GenerateJob.java
+++ b/cobigen-eclipse/cobigen-eclipse/src/com/devonfw/cobigen/eclipse/workbenchcontrol/handler/GenerateJob.java
@@ -103,10 +103,7 @@ public class GenerateJob implements IRunnableWithProgress {
         }
       });
     } catch (Throwable e) {
-      PlatformUIUtil.getWorkbench().getDisplay().syncExec(() -> {
-        ExceptionHandler.handle(e, HandlerUtil.getActiveShell(this.event));
-      });
-
+      ExceptionHandler.handle(e, HandlerUtil.getActiveShell(this.event));
     }
 
     MDC.remove(InfrastructureConstants.CORRELATION_ID);

--- a/cobigen/cobigen-core/src/main/java/com/devonfw/cobigen/impl/config/reader/ContextConfigurationReader.java
+++ b/cobigen/cobigen-core/src/main/java/com/devonfw/cobigen/impl/config/reader/ContextConfigurationReader.java
@@ -91,7 +91,7 @@ public class ContextConfigurationReader {
       }
     } else {
       Path subConfigRoot = configRoot.resolve(ConfigurationConstants.TEMPLATE_RESOURCE_FOLDER);
-      if (Files.exists(subConfigRoot)) {
+      if (Files.isDirectory(subConfigRoot)) {
         checkForConflict(subConfigRoot, contextFile);
       }
       this.contextFiles.add(contextFile);

--- a/cobigen/cobigen-core/src/main/java/com/devonfw/cobigen/impl/config/reader/ContextConfigurationReader.java
+++ b/cobigen/cobigen-core/src/main/java/com/devonfw/cobigen/impl/config/reader/ContextConfigurationReader.java
@@ -85,22 +85,35 @@ public class ContextConfigurationReader {
         }
 
       } else {
+        checkForConflict(configRoot, contextFile);
         this.contextFiles.add(contextFile);
-        // check if conflict with old and modular configuration exists
-
-        if (!loadContextFilesInSubfolder(configRoot).isEmpty())
-          throw new InvalidConfigurationException(contextFile,
-              "You are using an old configuration of the templates in addition to new ones. Please make sure this is not the case as both at the same time are not supported. For more details visit this wiki page: "
-                  + WikiConstants.WIKI_UPDATE_OLD_CONFIG);
 
       }
     } else {
+      Path subConfigRoot = configRoot.resolve(ConfigurationConstants.TEMPLATE_RESOURCE_FOLDER);
+      if (Files.exists(subConfigRoot)) {
+        checkForConflict(subConfigRoot, contextFile);
+      }
       this.contextFiles.add(contextFile);
     }
 
     this.contextRoot = configRoot;
 
     readConfiguration();
+  }
+
+  /**
+   * Checks if conflict with old and modular configuration exists
+   *
+   * @param configRoot Path to root directory of the configuration
+   * @param contextFile Path to context file of the configuration
+   */
+  private void checkForConflict(Path configRoot, Path contextFile) {
+
+    if (!loadContextFilesInSubfolder(configRoot).isEmpty())
+      throw new InvalidConfigurationException(contextFile,
+          "You are using an old configuration of the templates in addition to new ones. Please make sure this is not the case as both at the same time are not supported. For more details visit this wiki page: "
+              + WikiConstants.WIKI_UPDATE_OLD_CONFIG);
   }
 
   /**

--- a/cobigen/cobigen-core/src/test/java/com/devonfw/cobigen/unittest/config/reader/ContextConfigurationReaderTest.java
+++ b/cobigen/cobigen-core/src/test/java/com/devonfw/cobigen/unittest/config/reader/ContextConfigurationReaderTest.java
@@ -47,6 +47,7 @@ public class ContextConfigurationReaderTest extends AbstractUnitTest {
    * @throws InvalidConfigurationException if a conflict occurred
    *
    */
+  @Test
   public void testConflictConfiguration() throws InvalidConfigurationException {
 
     Throwable bothPresent = assertThrows(InvalidConfigurationException.class, () -> {

--- a/documentation/guide_dev_troubleshooting.asciidoc
+++ b/documentation/guide_dev_troubleshooting.asciidoc
@@ -31,7 +31,7 @@ ic (https://devon.s2-eu.capgemini.com/nexus/content/groups/public/): Not authori
 
 ==== Solution
 
-Please note the message Not authorized , ReasonPhrase: Unauthorized. -> [Help 2]! 
+Please note the message Not authorized , ReasonPhrase: Unauthorized. -> [Help 2]!
 
 1. Please check, that you run the command by using the console.bat or a similar console initialized with the IDE environment variables.
 2. Please check your corporate login in the variables-customized.bat to be correct (`DEVON_NEXUS_USER` and `DEVON_NEXUS_PASSWD`). Make sure, that you restart the console.bat you are working in after changing the variables-customized.bat. Same holds for eclipse instances running. Please restart to make the new values accessible.
@@ -48,7 +48,7 @@ To test changes implemented on the `cobigen-core` you have to follow the next pr
 4. Open a console and step into `cobigen/cobigen-eclipse`. Run `mvn clean package -Pp2-build-photon,p2-build-stable,p2-build-experimental`.
 5. On Eclipse, go to `cobigen/cobigen-eclipse` and double-click 'plugins.xml'. On the bottom, click on 'runtime' tab. On 'classpath', add a new library and choose the jars you have just installed.
 6. Refresh on Eclipse, press F5 on the `cobigen-eclipse` inside Eclipse.
-5. If you still see compilation errors: On Eclipse, right-click `cobigen/cobigen-eclipse` -> Maven -> Update projects. 
+5. If you still see compilation errors: On Eclipse, right-click `cobigen/cobigen-eclipse` -> Maven -> Update projects.
 
 == Issues with workspace when Oomph automatic updates don't work (especially for Indian colleagues)
 Executing `eclipse-cobigen-development.bat` file will open Eclipse with all the projects automatically imported. Oomph creates 'Working Sets' and set 'Top Level Elements' pointing to that working set. For Countries where proxy restricts Oomph to execute, we see no projects imported into project explorer/Navigator. Rather than trying manual import which later can give build issues we should follow below solution.
@@ -65,7 +65,7 @@ Build Issues could be like:
 ==== Solution
 In Eclipse, you can click the small downward arrow in the upper right corner of the Navigator/Project Explorer view and go to 'Top Level Elements' and point them to 'Projects'. This should show all the projects inside Project Explorer View. Also, Each plugin should point to respective branch.
 
-== Issue when testing Eclipse plugin by Running as Eclipse Application. 
+== Issue when testing Eclipse plugin by Running as Eclipse Application.
 Error message will be like:
 [source]
 ----
@@ -83,4 +83,15 @@ Delete or rename the runtime-`EclipseApplication` inside workspaces folder. Re- 
 3) org.eclipse.core.runtime.CoreException: Plug-in com.devonfw.cobigen.eclipse was unable to load class com.devonfw.cobigen.eclipse.workbenchcontrol.handler.XXXXHandler.
 ```
 
- 
+== Issue when running Eclipse SWT bot tests via github actions with custom test projects
+Error message on github actions will be like:
+[source]
+----
+org.eclipse.swtbot.swt.finder.widgets.TimeoutException: Timeout after: 15000 ms.: The widget (of type 'Button' and with mnemonic 'Finish' and with style 'SWT.PUSH') was not enabled.
+...
+at com.devonfw.cobigen.eclipse.test.common.utils.EclipseUtils.importExistingGeneralProject(EclipseUtils.java:101)
+----
+But on the local system the build and test process works fine.
+
+==== Solution
+If you add an Eclipse test project, make sure that you git force add the .project file too. Otherwise the importExistingGeneralProject method will fail.

